derive_builder_core/src/setter.rs

```diff
@@ -58,7 +60,7 @@ pub struct Setter<'a> {
 
 impl<'a> ToTokens for Setter<'a> {
     fn to_tokens(&self, tokens: &mut Tokens) {
-        if self.enabled {
+        if self.enabled || self.try_enabled {
```

Comment:
PS: You can either pull all shared variables out of the `if self.enabled {` statement, or just emit two `Setter` objects per field (as suggested in another comment).